### PR TITLE
2.0.0 RC0

### PR DESCRIPTION
**Pull request by holl-**
Φ-Flow version 2 is the first major update since public release.

It introduces a functional style for writing simulations and provides lots of utility to go along with it. The most fundamental addition is a new `Tensor` class that wraps native tensors and extends them by named and typed dimensions. This makes code simulation more generic and robust, and less prone to errors.

### New major features

General & UI

* API documentation with pdoc
* phi.verify() and verify.py
* Matplotlib fallback GUI
* ModuleViewer

Physics & Fields

* FLIP simulation functions and demo
* Improved PointCloud support
* Field I/O, batched Scenes
* Unified SampledFields

Math

* Profiling
* TensorFlow 2 with eager execution
* Tensors with named and typed dimensions
* Automatic conversion from NumPy
* Automatic dtype casting
* Automatic reshaping based on dimension type and name
* Sparse matrix generation
* PyTorch / TensorFlow parity
* Experimental Jax support
* Non-uniform tensors
* Delayed allocation and stacking
* Unified gradient computation
* Unified FP precision
* Unified extrapolations

### CODE REVIEW

**Review by thunil — APPROVED** [review]
Looks very good! My tests so far go through.
